### ISSUE TITLE
[Automation API] Expose plugin installation commands

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/pulumi#5080]
InlineSource doesn't use plugin acquisition code. Plugins need to be manually managed and we need to expose this functionality.

**Comment by leezen** [comment]
If I'm using inline source, wouldn't it be 'natural' to link the plugins into the binary? Is there a way to somehow turn that into part of the build process?

**Comment by EvanBoyle** [comment]
I don't think this is a trivial change. In the traditional CLI model, I believe the engine initiates this process over the RPC boundary relying on a binary in the pluginDir/path with the naming convention of `pulumi-resource-${provider}`. 

So we'd need either one of two things:

1. providers automatically do a `go install` somehow when the SDK is included in the program (package init functions maybe? but probably not).
2. redesign the way we load plugins so that they can be linked in dynamically.
